spec/setfrozentag_action_spec.rb

```diff
@@ -0,0 +1,102 @@
+require 'spec_helper'
+
+describe Fastlane::Actions::SetfrozentagAction do
+  let(:test_repository) { 'test-repository' }
+  let(:test_token) { 'Test-GithubToken-1234' }
+  let(:test_milestone) do
+    { title: '10.1', number: '1234' }
+  end
+  let(:client) do
+    instance_double(
+      Octokit::Client,
+      list_milestones: [test_milestone],
+      update_milestone: nil,
+      user: instance_double('User', name: 'test'),
+      'auto_paginate=': nil
+    )
+  end
+
+  before do
+    allow(Octokit::Client).to receive(:new).and_return(client)
+  end
+
+  it 'properly passes the environment variable `GITHUB_TOKEN` to Octokit::Client' do
+    ENV['GITHUB_TOKEN'] = test_token
+    expect(Octokit::Client).to receive(:new).with(access_token: test_token)
+    run_action_without_key(:github_token)
+  end
+
+  it 'properly passes the parameter `:github_token` all the way to Octokit::Client' do
+    expect(Octokit::Client).to receive(:new).with(access_token: test_token)
+    run_described_fastlane_action(default_params)
+  end
+
+  it 'prioritizes `:github_token` parameter over `GITHUB_TOKEN` environment variable if both are present' do
+    ENV['GITHUB_TOKEN'] = 'Test-EnvGithubToken-1234'
+    expect(Octokit::Client).to receive(:new).with(access_token: test_token)
+    run_described_fastlane_action(default_params)
+  end
```

Comment:
Should be able to DRY those, or at least group them in a `describe` given how all 3 are related